subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -49,13 +55,57 @@ static ssize_t smp_bt_chr_write(struct bt_conn *conn,
 				const void *buf, uint16_t len, uint16_t offset,
 				uint8_t flags)
 {
+#ifndef CONFIG_MCUMGR_SMP_PKASM_BT
+	int ret;
+
+	ret = zephyr_smp_pkasm_collect(&smp_bt_transport, buf, len);
+
+	/*
+	 * Collection can only fail due to being unable to allocate new buffer or while receiving
+	 * more data than expected in payload.
+	 */
+	if (in_progress == false && ret >= 0) {
+		/* On successful collection of first fragment of a new packet, transpor
+		 * information needs to be attached.
+		 */
+		struct smp_bt_user_data *ud = zephyr_smp_pkasm_get_ud(&smp_bt_transport);
+
+		if (ud != NULL) {
+			ud->conn = bt_conn_ref(conn);
+		}
+		in_progress = true;
+
+	} else if (ret < 0) {
+		/* Failed to collect buffer, discard packet and report error */
+		struct smp_bt_user_data *ud = zephyr_smp_pkasm_get_ud(&smp_bt_transport);
+
+		if (ud != NULL) {
+			bt_conn_unref(ud->conn);
```

Comment:
`bt_conn_unref`, should be followed by ud->conn = NULL;

Reply:
I can add that, but the re-assembly does not care about the contents of the ud and will simply release net_buf to the pool.

Reply:
Done